File: frontend/src/app/models/Repartition.ts
export class Repartition {
    user!: number;
    weight!: number;

    get user_id(): number { return this.user; }
    set user_id(v: number) { this.user = v; }

    constructor(userId: number, weight: number = 1) {
        this.user = userId;
        this.weight = weight;
    }
}

File: backend/Models/Entities/Repartition.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Tricount.Models.Entities;

public class Repartition {
    
    public int OperationId { get; set; }
    public Operation Operation { get; set; } = null!;

    public int UserId { get; set; }
    public User User { get; set; } = null!;

    public int Weight { get; set; }
}